frontend/public/components/image-stream.tsx

```diff
@@ -18,6 +18,42 @@ const menuActions = [
   ...common,
 ];
 
+// A valid specTag has a 'builder' tag. no 'hidden' tag, and exists in 'status.tags'.
+export const getBuilderSpecTagName = (imagestream) => {
+  let validSpecTag: any;
+
+  if (!imagestream.status) {
+    return null;
+  }
+
+  validSpecTag = _.find(imagestream.spec.tags, (specTag) => {
+    let specTagTags = _.get(specTag, 'annotations.tags');
+    if (!specTagTags) {
+      return false;
+    }
+    specTagTags = specTagTags.split(/\s*,\s*/);
+    if (!_.includes(specTagTags, 'builder') || _.includes(specTagTags, 'hidden')) {
+      return false;
+    }
+    return _.some(imagestream.status.tags, { tag: specTag.name });
+  });
```

Comment:
Hey @dtaylor113, I had mentioned that I already had something similar for the create from builder page. Here's what I have:

https://github.com/spadgett/console/blob/new-app/frontend/public/components/image-stream.tsx#L15-L39

We should try to implement this only once.

Reply:
I hope to have it ready for review either today or Monday.